src/ripple/nodestore/impl/Database.cpp

```diff
@@ -115,13 +115,13 @@ Database::asyncFetch(uint256 const& hash, std::uint32_t seq,
 }
 
 std::shared_ptr<NodeObject>
-Database::fetchInternal(uint256 const& hash, Backend& srcBackend)
+Database::fetchInternal(uint256 const& hash, std::shared_ptr<Backend> backend)
```

Comment:
The motivation is to keep the backend alive during the call, as it can now be released by another thread, making a reference obsolete. The public versions `asyncFetch`, `fetch`, `import` and `copyLedger` have documentation in the header file. Would comments tying the protected functions to the public be helpful?

Reply:
Thanks for confirmation on the `shared_ptr`.  Occasionally folks get excited about `shared_ptr` and use it when a `unique_ptr` would suffice.  But that's not happening here.  Sorry for the noise.

Yeah, adding short comments pointing to the public functions might help.  I'm reading from a very different perspective than yours.  You already know which functions are reflections of public methods; I didn't (and am unlikely to remember in the future).

Reply:
Will add comments, thanks!